src/core/js/models/adaptModel.js

```diff
@@ -287,24 +289,42 @@ define([
                 }
             }
 
-            searchChildren(children);
+            if (this._children === descendants) {
+                returnedDescendants = this.getChildren().models;
+            } else {
+                searchChildren(this.getChildren().models);
+            }
 
-            return returnedDescendants;
+            if (!options) {
+                return returnedDescendants;
+            }
+
+            if (options.where) {
+                return _.filter(returnedDescendants, function(descendant) {
+                    for (var property in options.where) {
```

Comment:
Are we particularly against using Underscore here? Should readability outweigh being performant?
```js
return _.findWhere(descendant, properties);
```

Reply:
[I did already have a bit of a look at this](https://github.com/adaptlearning/adapt_framework/issues/2058#issuecomment-386673476) and felt that on balance `for...in` was the better option